src/identify/index.js

```diff
@@ -172,8 +183,40 @@ class IdentifyService {
     // Get the observedAddr if there is one
     observedAddr = IdentifyService.getCleanMultiaddr(observedAddr)
 
+    // LEGACY: differentiate message with SignedPeerRecord
+    if (protocol === MULTICODEC_IDENTIFY_1_0_0) {
```

Comment:
No need to check the protocol, I'd just check the existence of `signedPeerRecord`. I think it might be better to catch and log the errors unmarshalling the signed peer record and then fallback to adding the `listenAddrs`. If for some reason the payload is malformed this will allow us to easily fallback.